src/AspNet.Security.OAuth.Weixin/WeixinAuthenticationHandler.cs

```diff
@@ -34,17 +32,18 @@ public class WeixinAuthenticationHandler : OAuthHandler<WeixinAuthenticationOpti
             : base(options, logger, encoder, clock)
         {
         }
+        private const string OauthState = "oauthstate";
         protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync()
         {
-            if (Options.AuthorizationEndpoint != WeixinAuthenticationDefaults.AuthorizationEndpoint)
+            if (!IsWeixinAuthorizationEndpointInUse())
             {
-                var oauthstate = Request.Query["oauthstate"];
-                Request.Query = new QueryCollection(Request.Query.Select(c =>
+                var state = Request.Query[OauthState];
+                var query = Request.Query.ToDictionary(c => c.Key, c => c.Value, StringComparer.OrdinalIgnoreCase);
+                if (query.ContainsKey("sata"))
```

Comment:
Use `TryGetValue()` - it's more efficent that `ContainsKey()` and `[]`.

Reply:
`if (Request.Query.TryGetValue(OauthState, out var state))
                {
                    var query = Request.Query.ToDictionary(c => c.Key, c => c.Value, StringComparer.OrdinalIgnoreCase);
                    if (query.TryGetValue("sata",out var _))
                    {
                        query["sata"] = state;
                        Request.QueryString = QueryString.Create(query);
                    }
                }`
So?